build-logic/buildquality/src/main/kotlin/gradlebuild.ci-reporting.gradle.kts

```
@@ -45,168 +34,13 @@ val testFilesCleanup = extensions.create<TestFileCleanUpExtension>("testFilesCle
- */
-fun tmpTestFiles(): Map<File, List<String>> = layout.buildDirectory.dir("tmp/test files").get().asFile
-    .listFiles()
-    ?.associateWith { dir ->
-        val dirPath = dir.toPath()
-        Files.walk(dirPath).use { paths ->
-            paths.filter { !it.toFile().isDirectory }
-                .limit(4)
-                .map { dirPath.relativize(it).toString() }
-                .collect(Collectors.toList())
-        }
-    }?.filter {
-        it.value.isNotEmpty()
-    } ?: emptyMap()
-
-fun executedTasks() = gradle.taskGraph.allTasks.filter { it.project == project && it.state.executed }
-
-fun failedTasks() = gradle.taskGraph.allTasks.filter { it.project == project && (it.state.failure != null || it.containsFailedTest()) }
-
-// We count the test task containing flaky result as failed
-fun Task.containsFailedTest(): Boolean {
-    if (this !is Test) {
-        return false
-    }
-
-    var containingFailures = false
-
-    val serializer = TestResultSerializer(binaryResultsDirectory.get().asFile)
-    if (serializer.isHasResults) {
-        serializer.read {
-            if (failuresCount > 0) {
-                containingFailures = true
-            }
-        }
-    }
-    return containingFailures
-}
-
-fun Task.failedTaskGenericHtmlReports() = when (this) {
-    is Reporting<*> -> listOf(this.reports["html"].outputLocation.get().asFile)
-    else -> emptyList()
-}
-
-fun Task.customReports() = when (this) {
-    is ValidatePlugins -> listOf(outputFile.get().asFile)
-    is Classycle -> listOf(reportFile.get().asFile)
-    is FindBrokenInternalLinks -> listOf(reportFile.get().asFile)
-    is DistributionTest -> listOf(
-        gradleInstallationForTest.gradleUserHomeDir.dir("test-kit-daemon").get().asFile,
-        gradleInstallationForTest.gradleUserHomeDir.dir("kotlin-compiler-daemon").get().asFile,
-        gradleInstallationForTest.daemonRegistry.get().asFile
-    )
-    else -> emptyList()
-}
-
-fun Task.attachedReportLocations() = when (this) {
-    is JapicmpTask -> listOf(richReport.destinationDir.resolve(richReport.reportName))
-    is PerformanceTest -> listOf(reportDir.parentFile)
-    else -> emptyList()
-}
-
-fun zip(destZip: File, srcDir: File) {
-    destZip.parentFile.mkdirs()
-    ZipOutputStream(FileOutputStream(destZip), StandardCharsets.UTF_8).use { zipOutput ->
-        val srcPath = srcDir.toPath()
-        Files.walk(srcPath).use { paths ->
-            paths
-                .filter { Files.isRegularFile(it, LinkOption.NOFOLLOW_LINKS) }
-                .forEach { path ->
-                    val zipEntry = ZipEntry(srcPath.relativize(path).toString())
-                    zipOutput.putNextEntry(zipEntry)
-                    Files.copy(path, zipOutput)
-                    zipOutput.closeEntry()
-                }
-        }
-    }
-}
+    val projectStates = objects.newInstance(TestFilesCleanupProjectState::class.java)
```

```suggestion
    val projectState = objects.newInstance(TestFilesCleanupProjectState::class.java)
```

Nit: This should be singular.